code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -286,15 +287,70 @@
 	var/datum/faction/malf/unction = find_active_faction_by_type(/datum/faction/malf)
 	if (!unction)
 		unction = ticker.mode.CreateFaction(/datum/faction/malf, null, 1)
-	var/mob/M = pick(candidates)
-	assigned += M
-	candidates -= M
+
+	var/mob/M = progressive_job_search()
+	if(M.mind.assigned_role != "AI")
+		for(var/mob/new_player/player in mode.candidates) //mode.candidates is everyone readied up, not to be confused with candidates
+			if(player.mind.assigned_role == "AI")
+				//We have located an AI to replace
+				displace_AI(player)
+				message_admins("Displacing AI played by: [key_name(player)].")
+	//There was no AI to displace, we're making one fresh
+	M.mind.assigned_role = "AI"
 	var/datum/role/malfAI/AI = new
 	AI.AssignToRole(M.mind,1)
-	unction.HandleRecruitedRole(AI)
-	AI.Greet(GREET_ROUNDSTART)
+	unction.HandleNewMind(AI)
+	AI.Greet()
 	return 1
 
+//First try to recruit an AI, then try to recruit a cyborg, then go to anybody left.
+/datum/dynamic_ruleset/roundstart/malf/proc/progressive_job_search()
+	message_admins("Beginning search for Malf AI candidate.")
+	for(var/job in job_priority)
+		for(var/mob/M in candidates)
+			if(M.mind.assigned_role == job)
+				assigned += M
+				candidates -= M
+				message_admins("Selected candidate with job [job].")
+				return M
+		message_admins("No valid candidates for job [job].")
+	message_admins("Selecting candidate from list: [json_encode(candidates)]")
+	var/mob/M = pick(candidates)
+	assigned += M
+	candidates -= M
+	return M
+
+/datum/dynamic_ruleset/roundstart/malf/proc/displace_AI(var/mob/new_player/old_AI)
+	old_AI.mind.assigned_role = null
+	var/list/shuffledoccupations = shuffle(job_master.occupations)
+	for(var/level = 1 to 3)
+		if(old_AI.mind.assigned_role)
+			break
+		for(var/datum/job/job in shuffledoccupations)
+			if(job_master.TryAssignJob(old_AI,level,job))
+				break
+	if(old_AI.mind.assigned_role)
+		return
+	if(old_AI.client.prefs.alternate_option == GET_RANDOM_JOB)
+		job_master.GiveRandomJob(old_AI)
+		return
+	else if(old_AI.client.prefs.alternate_option == BE_ASSISTANT)
+		job_master.AssignRole(old_AI, "Assistant")
+	else
+		to_chat(old_AI, "<span class='danger'>You have been returned to lobby due to your job preferences being filled.")
+		old_AI.ready = 0
+
+	/*old_AI.ready = 0
+	switch(alert(old_AI,"Your system has been corrupted by a malfunction, but you were not a valid candidate to be malfunctioning AI.", "AI Malfunction!",
```

Comment:
Will this stay commented?

Reply:
Good point, no reason not to just delete it